users/service.go

```diff
@@ -110,3 +142,32 @@ func (svc usersService) UserInfo(ctx context.Context, token string) (User, error
 		Metadata: dbUser.Metadata,
 	}, nil
 }
+
+func (svc usersService) GenerateResetToken(ctx context.Context, email, host string) error {
+	user, err := svc.users.RetrieveByID(ctx, email)
+	if err != nil || user.Email == "" {
+		return ErrUserNotFound
+	}
+
+	tok, err := token.Generate(email, 0)
+	if err != nil {
+		return ErrGeneratingResetToken
+	}
+
+	token.SendToken(host, email, tok)
```

Comment:
Think about using Tokenizer interface that is implemented by the tokenizer structure in the `token` package in order to be able to mock it.